src/frontends/tensorflow/src/input_model.cpp

```diff
@@ -170,82 +170,97 @@ std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::get_op_place
-            std::string output_port_name = producer_name + ":" + std::to_string(producer_output_port_idx);
-            if (m_tensor_places.find(output_port_name) != m_tensor_places.end()) {
-                const auto& tensor_place = m_tensor_places[output_port_name];
-                is_input = is_input || (tensor_place->is_input() ? true : false);
+    // the traversing algorithm to compute topologically sorted nodes is taken from topological_sort in
+    // core/graph_util.hpp
+    while (ops_to_do.size() > 0) {
+        auto current_operation_place = ops_to_do.top();
+        auto current_operation_decoder = current_operation_place->get_decoder();
+        auto current_operation_name = current_operation_decoder->get_op_name();
+        if (ops_done.count(current_operation_place) == 0) {
+            bool can_add = true;
+            auto input_count = current_operation_decoder->get_input_size();
+            for (size_t input_port_idx = 0; input_port_idx < input_count; ++input_port_idx) {
+                std::string producer_name;
+                size_t producer_output_port_idx;
+                try {
+                    current_operation_decoder->get_input_node(input_port_idx, producer_name, producer_output_port_idx);
+                } catch (const std::exception& e) {
+                    FRONT_END_THROW("[ ERROR ] Exception happened when preparing input " +
+                                    std::to_string(input_port_idx) + " for op '" +
+                                    current_operation_decoder->get_op_name() + "', expected input name: '" +
+                                    producer_name +
+                                    "', expected input port index: " + std::to_string(producer_output_port_idx) + '\n');
+                }
+
+                // skip conditional edges for all operators
+                if (is_conditional_edge(producer_name)) {
+                    continue;
+                }
+
+                // TODO: re-implement the logic below using Place graph structure (with OpPlace, In/OutPortPlace
+                // connections) and based on check if Place->is_input() decide to leave a node or not
+
+                // is_input is a flag to leave producer operation node or not.
+                // this producing node is not left if consumer is pruned by its input port,
+                // the producer node is pruned by its output port or the producer becomes new input
+                // 1. check if the current node is pruned by its input port
+                bool is_input = false;
+                std::string input_port_name = std::to_string(input_port_idx) + ":" + current_operation_name;
+                if (m_tensor_places.find(input_port_name) != m_tensor_places.end()) {
+                    const auto& tensor_place = m_tensor_places[input_port_name];
+                    is_input = is_input || (tensor_place->is_input() ? true : false);
```

Comment:
is_input() returns bool
```suggestion
                    is_input = is_input || tensor_place->is_input();
```